### PR TITLE
Providers can categorise disregarded benefit transactions (again)

### PR DESCRIPTION
**Pull request by cpjmcquillan**
This behaviour was erroneously removed in #4366.

Once a client has provided access to their transactions using TrueLayer, providers need to be able to categorise their bank transactions.

One of these categories is "Disregarded benefits".

In #4366, the `TransactionType.find_with_children` method was removed (alongside its use in `Providers::Means::IdentifyTypesOfIncomesController`.

What we didn't know then (but we do now) is that the intention of this method was to allow child transaction types to be associated to an application when their parent transaction type was identified.

For instance:

  When "Benefits" are released, we want to create "Disregarded benefits"
  transaction types too.

This updates the affected controller to create transaction type records for all identified credit transaction types including their children (but not housing benefit).

A feature test has been added, since this behaviour was previously untested --hence the regression!

### CODE REVIEW

File: features/providers/non_passported_journey.feature
Comment (cpjmcquillan) [inline features/providers/non_passported_journey.feature:85]: I want to uncomment this, but I'm not sure how to set up the appropriate VCR cassette so that state benefit data is returned 🤔

File: features/step_definitions/generic_steps.rb
Comment (cpjmcquillan) [inline features/step_definitions/generic_steps.rb:21]: I removed the `Regexp.quote` because it was confusing, and it didn't match on the income summary page.

Just matching on the provided string seems to work nicely for all the integration tests.

File: spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb
Comment (jsugarman) [inline spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb:5]: nit - linebreak

File: spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb
Comment (jsugarman) [inline spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb:12]: nit - linebreak

File: spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb
Comment (jsugarman) [inline spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb:18]: nit - linebreak

Comment (jsugarman) [reply spec/requests/providers/means/identify_types_of_incomes_controller_spec.rb:18]: Personally I'd like to see a request spec here as well. bit belt and braces with the feature test but prefer to have to much

```
    context "when benefits selected" do
      let(:transaction_type_ids) { [benefits.id] }
      let(:benefits) { create(:transaction_type, :benefits) }

      before do
        create(:transaction_type, :excluded_benefits, parent_id: benefits.id)
      end

      it "adds excluded_benefits transaction type as well as benefits" do
        expect { request }.to change(LegalAidApplicationTransactionType, :count).by(2)
        expect(legal_aid_application.reload.transaction_types.pluck(:name)).to match_array(%w[benefits excluded_benefits])
      end
    end
```

File: app/models/transaction_type.rb
Comment (jsugarman) [inline app/models/transaction_type.rb:53]: 👍 preferable to the old method i'd say

File: features/providers/non_passported_journey.feature
Comment (jsugarman) [reply features/providers/non_passported_journey.feature:85]: could pair on tomorrow if you like, for vcr i generally ensure the local machine config is pointing at relevant staging remote apps then delete the exist cassette and rerun feature test.

**Review by jsugarman — COMMENTED** [review]
looks good, 

- wary of removing the regex myself as i think it prevent false positives but no biggee (was it causing failures?)
- fairly sure this controller is only used for truelayer since EBSU was enabled  so we do not need to worry about disregarded benefits for bank statement upload flows 🤔 

And yes, UAT of truelayer and EBSU journeys as a minimum is necessary

File: app/models/transaction_type.rb
Comment (jsugarman) [reply app/models/transaction_type.rb:53]: i guess it avoids an unnecessary DB hit too so 👍

File: features/providers/non_passported_journey.feature
Comment (cpjmcquillan) [reply features/providers/non_passported_journey.feature:85]: Sounds good, I'll try and schedule something. 

I don't like Cucumber or VCR so it's like two nemeses colliding 😆

File: features/step_definitions/generic_steps.rb
Comment (cpjmcquillan) [reply features/step_definitions/generic_steps.rb:21]: Ok, that makes sense.

I'm happy to leave it if we can figure out why it wouldn't match `Benefits` on that page 🤔

Comment (jsugarman) [reply features/step_definitions/generic_steps.rb:21]: @cpjmcquillan am happy to approve this PR if we can just revert this change that removes the regex. Can you put it in a separate branch to discuss so it does not block this PR which we need to get in ASAP. I am just concerned that changing this generic step used in many other features could introduce false positives in future if we remove.

File: features/providers/non_passported_journey.feature
Comment (jsugarman) [inline features/providers/non_passported_journey.feature:77]: not a blocker but for this page I would use be tempted to use a more general

And I should not see 'Housing benefit'

File: features/step_definitions/generic_steps.rb
Comment (cpjmcquillan) [reply features/step_definitions/generic_steps.rb]: Sure 👍

**Review by jsugarman — APPROVED** [review]
🎖️

**Pull request merged**
